lib/src/main/java/grpcbridge/util/ProtoVisitor.java

```diff
@@ -1,13 +1,56 @@
 package grpcbridge.util;
 
-import static java.lang.String.format;
-
+import com.google.protobuf.BoolValue;
+import com.google.protobuf.BytesValue;
+import com.google.protobuf.Descriptors.Descriptor;
 import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.DoubleValue;
+import com.google.protobuf.FloatValue;
+import com.google.protobuf.Int32Value;
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.StringValue;
+import com.google.protobuf.UInt32Value;
+import com.google.protobuf.UInt64Value;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.lang.String.format;
 
 public abstract class ProtoVisitor {
     public enum SimpleFieldType {
         INT, LONG, BOOL, DOUBLE, FLOAT, STRING, BYTES, ENUM;
 
+        /**
+         * Maps supported wrapper types to a {@link SimpleFieldType}.
+         *
+         * There's no simple way to identify whether the message is a wrapper, we have to use
+         * explicit descriptors matching.
+         */
+        static Map<Descriptor, SimpleFieldType> wrappers = new HashMap<Descriptor, SimpleFieldType>() {
```

Comment:
Can we combine this with the definition in Variable? Call it SimpleFieldMapper?

Reply:
Combined into `SimpleFieldMapper`.